fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java

```diff
@@ -1325,4 +1332,596 @@ public void testCreateAclWithAccessToClassForBinary() throws Exception {
+                "DELETE { <> ldp:membershipResource ?o } \n" +
+                "INSERT { <> ldp:membershipResource <" + targetResource + "> } \n" +
+                "WHERE { <> ldp:membershipResource ?o }";
+        patchDirect.setEntity(new StringEntity(patch_text, sparqlContentType));
+        assertEquals(HttpStatus.SC_NO_CONTENT, getStatus(patchDirect));
+
+        // Delete the ldp:membershipRelation and add it with INSERT DATA {}
+        final String patch_delete_relation = "prefix ldp: <http://www.w3.org/ns/ldp#> \n" +
+                "DELETE DATA { <> ldp:membershipResource <" + targetResource + "> }";
+        final HttpPatch patchDirect2 = new HttpPatch(directUri);
+        setAuth(patchDirect2, username);
+        patchDirect2.setEntity(new StringEntity(patch_delete_relation, sparqlContentType));
+        assertEquals(HttpStatus.SC_NO_CONTENT, getStatus(patchDirect2));
+
+        final String patch_insert_relation = "prefix ldp: <http://www.w3.org/ns/ldp#> \n" +
+                "INSERT DATA { <> ldp:membershipResource <" + targetResource + "> }";
+        final HttpPatch patchDirect3 = new HttpPatch(directUri);
+        setAuth(patchDirect3, username);
+        patchDirect3.setEntity(new StringEntity(patch_insert_relation, sparqlContentType));
+        assertEquals(HttpStatus.SC_NO_CONTENT, getStatus(patchDirect3));
+
+        // Ensure we can write to the writeable resource.
+        testCanWrite(writeableResource, username);
+    }
+
+    /**
+     * Utility function to ingest a ACL from a string.
+     *
+     * @param resourcePath Path to the resource if doesn't end with "/fcr:acl" it is added.
+     * @param acl the text/turtle ACL as a string
+     * @param username user to ingest as
+     * @return
+     * @throws IOException on StringEntity encoding or client execute
+     */
+    private HttpResponse ingestAclString(final String resourcePath, final String acl, final String username)
+            throws IOException {
+        final String aclPath = (resourcePath.endsWith("/fcr:acl") ? resourcePath : resourcePath + "/fcr:acl");
+        final HttpPut putReq = new HttpPut(aclPath);
+        setAuth(putReq, username);
+        putReq.setHeader("Content-type", "text/turtle");
+        putReq.setEntity(new StringEntity(acl, turtleContentType));
+        return execute(putReq);
+    }
+
+    /**
+     * Ensure that a writeable resource is still writeable
+     *
+     * @param writeableResource the URI of the writeable resource.
+     * @param username the user will write access.
+     */
+    private void testCanWrite(final String writeableResource, final String username) {
+        // Try to create a basic container inside the writeable resource with POST.
+        final HttpPost okPost = postObjMethod(writeableResource);
+        setAuth(okPost, username);
+        assertEquals(HttpStatus.SC_CREATED, getStatus(okPost));
+
+        // Try to create a basic container inside the writeable resource with PUT.
```

Comment:
Would it be even more revealing to test a `PATCH` of properties on the container vs another "basic container" child creation?